website/docs/r/search_index.html.markdown

```diff
@@ -0,0 +1,243 @@
+---
+layout: "mongodbatlas"
+page_title: "MongoDB Atlas: search index"
+sidebar_current: "docs-mongodbatlas-resource-search-index"
+description: |-
+Provides a Search Index resource.
+---
+
+# mongodbatlas_search_index
+
+`mongodbatlas_search_index` provides a Search Index resource. This allows indexes to be created.
+
+## Example Usage
+
+```hcl
+resource "mongodbatlas_search_index" "test" {
+  name   = "project-name"
+  project_id = "<PROJECT_ID>"
+  cluster_name = "<CLUSTER_NAME>"
+  
+  analyzer = "lucene.standard"
+  collectionName = "collection_test"
+  database = "database_test"
+  mappings_dynamic = true
+  
+  searchAnalyzer = "lucene.standard"
+}
+```
+
+## Argument Reference
+
+* `name` - (Required) The name of the search index you want to create.
+* `project_id` - (Required) The ID of the organization or project you want to create the search index within.
+* `cluster_name` - (Required) The name of the cluster where you want to create the search index within.
+
+
+
+~> **NOTE:** Atlas limits the number of users to a maximum of 100 teams per project and a maximum of 250 teams per organization.
+* `analyzer` - [Analyzer](https://docs.atlas.mongodb.com/reference/atlas-search/analyzers/#std-label-analyzers-ref) to use when creating the index. Defaults to [lucene.standard](https://docs.atlas.mongodb.com/reference/atlas-search/analyzers/standard/#std-label-ref-standard-analyzer)
+
+* `analyzers` - [Custom analyzers](https://docs.atlas.mongodb.com/reference/atlas-search/analyzers/custom/#std-label-custom-analyzers) to use in this index (this is an array of objects).
+
+* `collection_name` - (Required) Name of the collection the index is on.
+
+* `database` - (Required) Name of the database the collection is in.
+
+* `mappings_dynamic` - Indicates whether the index uses dynamic or static mapping. For dynamic mapping, set the value to `true`. For static mapping, specify the fields to index using `mappings_fields`
+
+* `search_analyzer` - [Analyzer](https://docs.atlas.mongodb.com/reference/atlas-search/analyzers/#std-label-analyzers-ref) to use when searching the index. Defaults to [lucene.standard](https://docs.atlas.mongodb.com/reference/atlas-search/analyzers/standard/#std-label-ref-standard-analyzer)
+
+### Mapping fields
+`mapping_fields` attribute is required when `mappings_dynamic` is true. This field needs to be a JSON string in order to be decoded correctly
+
+### Analyzers
```

Comment:
I am utterly baffled by how the user would maintain this part in Terraform - the way this looks they would create the analyzer here and then try to update it in another resource?   Can you please explain this?

I realize in reading this that this is more complicated than many of our resources have been due to the custom analyzers and may need some out of the normal process thinking.    @thetonymaster  We may need to meet on how this currently works and how it should work.

Reply:
Total misunderstanding on my part.